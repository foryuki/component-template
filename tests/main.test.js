const sao = require('sao')
const path = require('path')

const template = {
  fromPath: path.join(__dirname, '..')
}

describe('test', () => {
  it('ok', async () => {
    const stream = await sao.mockPrompt(template, {
      name: 'stub',
      username: 'placeholder',
      email: 'placholder@mymail.com'
    })

    expect(stream.fileList).toMatchSnapshot()

    const pkg = JSON.parse(stream.fileContents('package.json'))
    expect(pkg.name).toBe('or-stub')
    expect(pkg.author).toBe('placeholder <placholder@mymail.com>')
    expect(pkg.license).toBe('MIT')

    expect(stream.fileContents('readme.md')).toMatchSnapshot()
    expect(stream.fileContents('.babelrc')).toMatchSnapshot()
    expect(stream.fileContents('.npmrc')).toMatchSnapshot()
  })
})
