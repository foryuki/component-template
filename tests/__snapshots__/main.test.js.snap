// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ok 1`] = `
Array [
  ".babelrc.js",
  ".circleci/config.yml",
  ".editorconfig",
  ".gitignore",
  ".npmignore",
  ".npmrc",
  ".prettierrc",
  ".travis.yml",
  ".vscode/launch.json",
  "LICENSE",
  "commitlint.config.js",
  "examples/.npmrc",
  "examples/.storybook/addons.js",
  "examples/.storybook/config.js",
  "examples/.storybook/manager-head.html",
  "examples/.storybook/webpack.config.js",
  "examples/package.json",
  "examples/public/.gitkeep",
  "examples/stories/example.tsx",
  "examples/stories/index.story.tsx",
  "examples/stories/styles.scss",
  "examples/stories/util.tsx",
  "examples/tsconfig.json",
  "examples/types/declarations.d.ts",
  "greenkeeper.json",
  "jest.config.js",
  "package.json",
  "readme.md",
  "src/Sample.tsx",
  "src/index.ts",
  "src/styles.scss",
  "tests/Sample.test.tsx",
  "tests/__mocks__/fileMock.js",
  "tests/__mocks__/styleMock.js",
  "tests/setup.ts",
  "tsconfig.json",
  "tslint.json",
]
`;

exports[`test ok: .babelrc.js 1`] = `
"const presets = [
  [
    '@babel/preset-env',
    {
      modules: process.env.NODE_ENV === 'es' ? false : 'commonjs'
    }
  ],
  '@babel/preset-react',
  '@babel/preset-typescript'
]

const plugins = [
  '@babel/plugin-transform-runtime',
  '@babel/plugin-proposal-class-properties',
  '@babel/plugin-proposal-object-rest-spread'
]

if (process.env.NODE_ENV === 'test') {
  plugins.push('@babel/plugin-transform-modules-commonjs')
}

module.exports = { presets, plugins }
"
`;

exports[`test ok: readme.md content 1`] = `
"# One React Component: stub


<p align=\\"center\\"><img width=\\"150\\" src=\\"https://cdn.jsdelivr.net/gh/one-react/assets/logo%402x.png\\" alt=\\"logo\\"></p>

[![TravisCI Build](https://img.shields.io/travis/one-react/stub.svg)](https://travis-ci.org/one-react/stub)
[![CircieCI Build](https://img.shields.io/circleci/project/github/one-react/stub.svg)](https://circleci.com/gh/one-react/stub)
[![Coverage](https://img.shields.io/codecov/c/github/one-react/stub.svg)](https://codecov.io/gh/one-react/stub) 
[![Version](https://img.shields.io/npm/v/or-stub.svg)](https://www.npmjs.com/package/or-stub)
[![Chat](https://img.shields.io/gitter/room/one-react-org/Lobby.svg)](https://gitter.im/one-react-org/Lobby)
[![License: MIT](https://img.shields.io/badge/License-MIT-brightgreen.svg)](https://opensource.org/licenses/MIT)
[![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)
[![Greenkeeper badge](https://badges.greenkeeper.io/one-react/stub.svg)](https://greenkeeper.io/) 

## Installation
\`\`\`
// with npm
npm install or-stub

// with yarn
yarn add or-stub

\`\`\`

## Usage
- Config webpack \`sass-loader\` if you are using webpack.

\`\`\`js
// webpack.config.js
{
  test: /\\\\.scss$/,
  use: [
    'style-loader',
    'css-loader',
    'sass-loader'
  ],
  include: [
    /node_modules\\\\/or\\\\-\\\\w+/ //include or-components
  ]
}
\`\`\`

## Basic Example

## API
\`\`\`ts
interface Props {
  
}
\`\`\`

## Customize Theme
**Customize in webpack**

The following variables in or-stub can be overridden:

\`\`\`scss
$or-btn-primary-color: $or-primary-color !default;
...
\`\`\`
For more variables, see [here](https://github.com/one-react/stub/blob/master/src/styles.scss).

Alternatively, you can override variables from [or-theme](https://github.com/one-react/theme/blob/master/src/variables.scss) to keep all or-components in a unified theme style.

First you should create a \`theme.scss\` file to declare the variables you want to override.

Then use the [data](https://github.com/webpack-contrib/sass-loader#environment-variables)  option provided by \`sass-loader\` to override the default values of the variables.

We take a typical \`webpack.config.js\` file as example to customize it's sass-loader options.

\`\`\`js
// webpack.config.js
{
  test: /\\\\.scss$/,
  use: [
    'style-loader',
    'css-loader',
    {
      loader: 'sass-loader',
      options: {
        data: fs.readFileSync(path.resolve(__dirname, 'theme.scss')) // pass theme.scss to sass-loader
      }
    }
  ],
  include: [
    /node_modules\\\\/or\\\\-\\\\w+/ //include or-components
  ]
}
\`\`\`

## Demos and Docs
> powered by [storybook](https://storybook.js.org/)

[Click Here](https://one-react.github.io/stub)

## License

MIT &copy; placeholder
"
`;

exports[`test ok: tsconfig.json 1`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": \\"esnext\\",
    \\"module\\": \\"commonjs\\",
    \\"baseUrl\\": \\"./src\\",
    \\"declarationDir\\": \\"./lib\\",
    \\"declaration\\": true,
    \\"allowSyntheticDefaultImports\\": true,
    \\"esModuleInterop\\": true,
    \\"jsx\\": \\"react\\",
    \\"moduleResolution\\": \\"node\\",
    \\"noUnusedLocals\\": true,
    \\"noUnusedParameters\\": true,
    \\"pretty\\": true,
    \\"skipLibCheck\\": true,
    \\"sourceMap\\": true,
    \\"lib\\": [
      \\"esnext\\",
      \\"es2015\\",
      \\"dom\\"
    ]
  },
  \\"include\\": [
    \\"src\\"
  ]
}
"
`;
